### ISSUE TITLE
New rule: Ensure module exports/returns something

### ISSUE THREAD

**Issue by platinumazure** [cvisco/eslint-plugin-requirejs#100]
More often than I care to admit, I've written code like the following:

```js
define(["some/dep"], function (Dep) {
    var MyThing = {
        // 500+ lines of code
    };

    // oops, forgot to return MyThing
});
```

We could create a rule which uses code path analysis to ensure that all code paths either return something (AMD) or assign to module or exports (CommonJS).

**Comment by cvisco** [comment]
Yeah, this makes sense. I guess we have rules that forbid certain _styles_ of return, but nothing that ensures we actually return something.

**Comment by cvisco** [comment]
Also, congrats on filing issue #100 :)
